### PR TITLE
Open file icon

### PR DESCRIPTION
**Pull request by allender**
icons for opening/diffing file from resource view in SCM panel.  Addresses https://github.com/mjcrouch/vscode-perforce/issues/63

### CODE REVIEW

**Comment by allender** [comment]
my git foo is rough.  I hope that I did this PR correctly.

**Comment by mjcrouch** [comment]
Thanks, the diff looks ok - I'm not sure exactly how you got different commit IDs for all the commits 
you merged from my repo, but it shouldn't make a difference as I rebase everything in to one commit anyway.
I'll have another look at it later to check the icon layout etc :)

File: resources/icons/light/diff.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path fill-rule="evenodd" clip-rule="evenodd" d="M2 3.5L2.5 3H7.5L8 3.5V12.5L7.5 13H2.5L2 12.5V3.5ZM3 12H7V6H3V12ZM3 5H7V4H3V5ZM9.5 3H14.5L15 3.5V12.5L14.5 13H9.5L9 12.5V3.5L9.5 3ZM10 12H14V10H10V12ZM10 8H14V4H10V8Z" fill="#C5C5C5"/>
```
Comment (mjcrouch) [inline resources/icons/light/diff.svg]: These are a bit hard to see on a light theme - would be good to match the revert icon - I think this is right:

image

```suggestion
<path fill-rule="evenodd" clip-rule="evenodd" d="M2 3.5L2.5 3H7.5L8 3.5V12.5L7.5 13H2.5L2 12.5V3.5ZM3 12H7V6H3V12ZM3 5H7V4H3V5ZM9.5 3H14.5L15 3.5V12.5L14.5 13H9.5L9 12.5V3.5L9.5 3ZM10 12H14V10H10V12ZM10 8H14V4H10V8Z" fill="#424242"/>
```

File: resources/icons/light/go-to-file.svg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path fill-rule="evenodd" clip-rule="evenodd" d="M8.06065 3.85356L5.91421 6L5.2071 5.29289L6.49999 4H3.5C3.10218 4 2.72064 4.15804 2.43934 4.43934C2.15804 4.72065 2 5.10218 2 5.5C2 5.89783 2.15804 6.27936 2.43934 6.56066C2.72064 6.84197 3.10218 7 3.5 7H4V8H3.5C2.83696 8 2.20107 7.73661 1.73223 7.26777C1.26339 6.79893 1 6.16305 1 5.5C1 4.83696 1.26339 4.20108 1.73223 3.73224C2.20107 3.2634 2.83696 3 3.5 3H6.49999L6.49999 3H6.49996L6 2.50004V2.50001L5.2071 1.70711L5.91421 1L8.06065 3.14645L8.06065 3.85356ZM5 6.50003L5.91421 7.41424L6 7.32845V14H14V7H10V3H9.06065V2.73227L8.32838 2H11.2L11.5 2.1L14.9 5.6L15 6V14.5L14.5 15H5.5L5 14.5V9.00003V6.50003ZM11 3V6H13.9032L11 3Z" fill="#C5C5C5"/>
```
Comment (mjcrouch) [inline resources/icons/light/go-to-file.svg]:
```suggestion
<path fill-rule="evenodd" clip-rule="evenodd" d="M8.06065 3.85356L5.91421 6L5.2071 5.29289L6.49999 4H3.5C3.10218 4 2.72064 4.15804 2.43934 4.43934C2.15804 4.72065 2 5.10218 2 5.5C2 5.89783 2.15804 6.27936 2.43934 6.56066C2.72064 6.84197 3.10218 7 3.5 7H4V8H3.5C2.83696 8 2.20107 7.73661 1.73223 7.26777C1.26339 6.79893 1 6.16305 1 5.5C1 4.83696 1.26339 4.20108 1.73223 3.73224C2.20107 3.2634 2.83696 3 3.5 3H6.49999L6.49999 3H6.49996L6 2.50004V2.50001L5.2071 1.70711L5.91421 1L8.06065 3.14645L8.06065 3.85356ZM5 6.50003L5.91421 7.41424L6 7.32845V14H14V7H10V3H9.06065V2.73227L8.32838 2H11.2L11.5 2.1L14.9 5.6L15 6V14.5L14.5 15H5.5L5 14.5V9.00003V6.50003ZM11 3V6H13.9032L11 3Z" fill="#424242"/>
```

**Comment by allender** [comment]
changes ok with me.  I pulled those icons directly from the vscode icon repo but yeah, they looked too light.

One thing I did notice is that if you change the SCM File Changes boolean, you have to refresh the source control provider before the icon will change properly on the resource.  I think this is because these things are tied to the resoruces when they are retrieved from perforce and when we change the option, those items still point to the old behaviors.  I'm not sure how to fix this properly

**Comment by mjcrouch** [comment]
looks better! Thanks

When I do it, the icon changes correctly, but the behaviour of clicking on the resource does not change until the scm provider is refreshed.
It uses the `get command()` property in `Resource` 
I believe vscode takes a snapshot when you fire a particular event from the scm provider to say that it has been updated - so we could fire that event, but we would have to use a config watcher to do that (I have one already, which could be used, or we could split that up a bit - been meaning to update do the same to apply the file watcher changes without a restart as well)

Alternatively, we could call a different command that checks the config and then calls out to the right command

Regardless that's probably out of scope here because it's a very minor issue, but you're welcome to look into it if you feel like it

**Pull request merged**